Enable recording warnings per AWS config node

haverma opened the pull request:
`ConvertConfigurationResult` class supported a map of `Configurations` but accepted only a single `Warnings` object, changed that to a `Map` of `<Hostname, Warnings>` so that we can store a separate `Warnings` object for each configuration. It worked for physical `Configuration` nodes as in that case the `configurations` map contains just a single node at a time, whereas for AWS configs it contains all AWS nodes parsed.

dhalperi commented:
Review status: 0 of 16 files reviewed at latest revision, 1 unresolved discussion.

---

*[projects/batfish/src/main/java/org/batfish/job/ConvertConfigurationResult.java, line 21 at r1](https://reviewable.io:443/reviews/batfish/batfish/1076#-L7yqTMPy35mxEjoHaFP:-L7yqTMQv0gw5HbI2xVZ:b9x5hv9) ([raw file](https://github.com/batfish/batfish/blob/8ba7bedc324199a7303cbf4f2fb31fd7dcc71d44/projects/batfish/src/main/java/org/batfish/job/ConvertConfigurationResult.java#L21)):*
> ```Java
>   private String _name;
> 
>   private Map<String, Warnings> _warningsByHost;
> ```

This is not going to be good enough. We need filename, not just hostname, etc.

What's the goal? Can you point me at the design doc?

There's already work started in this area....

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/batfish/batfish/1076)*
<!-- Sent from Reviewable.io -->

dhalperi commented:
After discussion, <img class="emoji" title=":LGTM:" alt=":LGTM:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>. A few cleanups remaining, but this seems like a solution that generally makes sense as a stopgap.

---

Reviewed 16 of 16 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[projects/batfish-common-protocol/src/main/java/org/batfish/common/Warnings.java, line 168 at r1](https://reviewable.io:443/reviews/batfish/batfish/1076#-L8-haWVPeDZln5EXJHf:-L8-haWWNSqOZycOHI1F:be2e20w) ([raw file](https://github.com/batfish/batfish/blob/8ba7bedc324199a7303cbf4f2fb31fd7dcc71d44/projects/batfish-common-protocol/src/main/java/org/batfish/common/Warnings.java#L168)):*
> ```Java
>   }
> 
>   public Warnings(Warnings other) {
> ```

I don't think this is a good method to add. It doesn't really clone warnings, it just copies settings from the other warnings.

The logic you really want is in Batfish.buildWarnings(). Move that logic as a public static helper method (`public static Warnings buildWarnings(Settings settings)`) somewhere and then use it in place of the existing builder.

Since CCJ has access to `Settings`, it should be able to call this helper to build a new warnings from Settings rather than copying from an existing warning.

---

*[projects/batfish/src/main/java/org/batfish/representation/aws/Region.java, line 265 at r1](https://reviewable.io:443/reviews/batfish/batfish/1076#-L8-iz3KvTEkh1sYs9Pa:-L8-iz3KvTEkh1sYs9Pb:b-29bru0) ([raw file](https://github.com/batfish/batfish/blob/8ba7bedc324199a7303cbf4f2fb31fd7dcc71d44/projects/batfish/src/main/java/org/batfish/representation/aws/Region.java#L265)):*
> ```Java
>     updateAllocatedIps();
>     for (Vpc vpc : getVpcs().values()) {
>       Warnings warnings = new Warnings(awsConfiguration.getWarnings());
> ```

to be replaced by `buildWarnings(settings)` or something.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/batfish/batfish/1076#-:-L8-jHZS2e7d0U1tk8cn:bmbqd98)*
<!-- Sent from Reviewable.io -->

haverma commented:
addressed comments, and did some refactoring to remove warnings instance member from `ConvertConfigurationJob` and `AwsConfiguration`

---

Review status: 12 of 17 files reviewed at latest revision, 2 unresolved discussions.

---

*[projects/batfish/src/main/java/org/batfish/representation/aws/Region.java, line 265 at r1](https://reviewable.io:443/reviews/batfish/batfish/1076#-L8-iz3KvTEkh1sYs9Pa:-L81AedOn2IsqcGuBmSO:bb7f2mf) ([raw file](https://github.com/batfish/batfish/blob/8ba7bedc324199a7303cbf4f2fb31fd7dcc71d44/projects/batfish/src/main/java/org/batfish/representation/aws/Region.java#L265)):*
<details><summary><i>Previously, dhalperi (Dan Halperin) wrote…</i></summary><blockquote>

to be replaced by `buildWarnings(settings)` or something.
</blockquote></details>

done

---

*[projects/batfish-common-protocol/src/main/java/org/batfish/common/Warnings.java, line 168 at r1](https://reviewable.io:443/reviews/batfish/batfish/1076#-L8-haWVPeDZln5EXJHf:-L81APPfTiDYsKty22XD:b-cwowlr) ([raw file](https://github.com/batfish/batfish/blob/8ba7bedc324199a7303cbf4f2fb31fd7dcc71d44/projects/batfish-common-protocol/src/main/java/org/batfish/common/Warnings.java#L168)):*
<details><summary><i>Previously, dhalperi (Dan Halperin) wrote…</i></summary><blockquote>

I don't think this is a good method to add. It doesn't really clone warnings, it just copies settings from the other warnings.

The logic you really want is in Batfish.buildWarnings(). Move that logic as a public static helper method (`public static Warnings buildWarnings(Settings settings)`) somewhere and then use it in place of the existing builder.

Since CCJ has access to `Settings`, it should be able to call this helper to build a new warnings from Settings rather than copying from an existing warning.
</blockquote></details>

Removed, made Batfish.buildWarning(Settings settings) as public static

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/batfish/batfish/1076#-:-L81Ai4xMKFBAqflngpa:b-i6sm79)*
<!-- Sent from Reviewable.io -->

dhalperi commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 6 of 6 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/batfish/batfish/1076#-:-L81Fis-LVhCWN-gjq61:bnfp4nl)*
<!-- Sent from Reviewable.io -->
